### PR TITLE
Apb 598 Check if agent is already subscribed, or has other enrolments

### PR DESCRIPTION
**Pull request by bravesirrobin81**

### CODE REVIEW

File: it/uk/gov/hmrc/agentsubscriptionfrontend/stubs/AuthStub.scala [hunk 1]
```diff
@@ -71,6 +71,42 @@ object AuthStub {
...
   }
 
+  def isSubscribedToMtd(user: SampleUser): Unit = {
+    stubFor(get(urlEqualTo(user.enrolmentsLink))
+        .willReturn(
+          aResponse()
+              .withStatus(200)
+              .withBody(
+                s"""
+                   |[{"key":"HMRC-AS-AGENT","identifiers":[{"key":"Arn","value":"ARN0001"}],"state":"Activated"}]
```
Comment (frankoid) [inline it/uk/gov/hmrc/agentsubscriptionfrontend/stubs/AuthStub.scala]: where does the `Arn` value for identifier key come from? Have you seen the service definition (Q Pack) for HMRC-AS-AGENT?

Comment (bravesirrobin81) [inline it/uk/gov/hmrc/agentsubscriptionfrontend/stubs/AuthStub.scala]: That's a very good question--I think it was originally a guess as we didn't have the q-pack. It works integrated with auth, which is probably why I didn't question it further---the changes to gg-stubs didn't include an identifier key for the enrolment at all.

Comment (bravesirrobin81) [inline it/uk/gov/hmrc/agentsubscriptionfrontend/stubs/AuthStub.scala]: And the reason it works is that we check only the presence of the enrolment, and not any identifiers.

Comment (frankoid) [inline it/uk/gov/hmrc/agentsubscriptionfrontend/stubs/AuthStub.scala]: Perhaps we can remove it until we know what the key will be? I'm looking at the code now to see whether it is actually needed.

Comment (frankoid) [inline it/uk/gov/hmrc/agentsubscriptionfrontend/stubs/AuthStub.scala]: Seems fine without it (tests pass), I've pushed a commit that removes it

**Comment by frankoid** [comment]
Looks like we don't need the "has other enrolments" check (requirements change) but we do still want the "already subscribed" check so I'm still going to merge this to get the refactorings and already subscribed check if you think that's a good idea @bravesirrobin81. Once the requirements change is confirmed we can remove the "has other enrolments" check.

**Comment by bravesirrobin81** [comment]
Fine by me.

**Pull request merged**